native/src/main/java/io/ballerina/lib/data/xmldata/utils/DataUtils.java

```diff
@@ -895,6 +893,30 @@ private static String getPrintableErrorMsg(BError err) {
         return errorMsg;
     }
 
+    public static boolean isEqualQualifiedName(QualifiedName firstQName, QualifiedName secondQName) {
+        if (firstQName == null || secondQName == null) {
+            return false;
+        }
+
+        if (firstQName.equals(secondQName)) {
+            return true;
+        }
+
+        return firstQName.getLocalPart().equals(secondQName.getLocalPart())
+                && (firstQName.getNamespaceURI().equals(Constants.NS_ANNOT_NOT_DEFINED)
+                || secondQName.getNamespaceURI().equals(Constants.NS_ANNOT_NOT_DEFINED));
+    }
+
+    public static boolean isSimpleType(Type type) {
+        return switch (type.getTag()) {
+            case TypeTags.JSON_TAG, TypeTags.ANYDATA_TAG, TypeTags.MAP_TAG, TypeTags.OBJECT_TYPE_TAG,
+                    TypeTags.RECORD_TYPE_TAG, TypeTags.XML_TAG -> false;
```

Comment:
is record type consider as a simple type?

Reply:
Had a discussion with @prakanth97 and seems it is ok

Reply:
We consider simple-type as non-complex type hence here record is not simple type